### PR TITLE
Add SPV_AMDX_shader_enqueue

### PR DESCRIPTION
**Pull request by KonstantinSeurer**
Based on https://github.com/KhronosGroup/Vulkan-Docs/blob/main/proposals/VK_AMDX_shader_enqueue.adoc .

### CODE REVIEW

File: include/spirv/unified1/spirv.core.grammar.json [hunk 1]
```diff
@@ -14484,6 +14638,13 @@
           "extensions" : [ "SPV_KHR_shader_clock" ],
           "version" : "None"
         },
+        {
+          "enumerant" : "ShaderEnqueueAMDX",
+          "value" : 5067,
+          "capabilities" : [ "ShaderEnqueueAMDX" ],
```
Comment (yavn) [inline include/spirv/unified1/spirv.core.grammar.json:14644]:
```suggestion
          "capabilities" : [ "Shader" ],
```

**Review by yavn — CHANGES_REQUESTED** [review]
Hi @KonstantinSeurer, thank you for the PR! We should have pushed it out with the release, but it was overlooked. Looks good, just a few minor corrections.

**Comment by yavn** [comment]
NB: For practical use in shaders, the following SPIR-V intrinsics helpers are provided:

- [shaderEnqueueSpirvIntrinsics.glsl.h](https://github.com/GPUOpen-LibrariesAndSDKs/Vulkan-Samples/blob/main/shaders/gpu_dispatch/glsl/shaderEnqueueSpirvIntrinsics.glsl.h)
- [shaderEnqueueSpirvIntrinsics.hlsl.h](https://github.com/GPUOpen-LibrariesAndSDKs/Vulkan-Samples/blob/main/shaders/gpu_dispatch/hlsl/shaderEnqueueSpirvIntrinsics.hlsl.h)

**Review by yavn — APPROVED** [review]

**Comment by johnkslang** [comment]
Why does this show a force pushes to main?

**Comment by KonstantinSeurer** [comment]
> Why does this show a force pushes to main?

Because I force pushed fixups to my main branch (https://github.com/KonstantinSeurer/SPIRV-Headers/commits/main)

**Comment by johnkslang** [comment]
Okay, thanks.

File: include/spirv/unified1/spirv.core.grammar.json [hunk 2]
```diff
@@ -4921,6 +4921,44 @@
...
     },
+    {
+      "opname" : "OpFinalizeNodePayloadsAMDX",
+      "class"  : "Reserved",
+      "opcode" : 5075,
+      "operands" : [
+        { "kind" : "IdRef", "name": "'Payload Array'" }
+      ],
+      "capabilities" : [ "ShaderEnqueueAMDX" ],
+      "extensions" : [ "SPV_AMDX_shader_enqueue" ],
```
Comment (alan-baker) [inline include/spirv/unified1/spirv.core.grammar.json:4932]: Only capabilities introduced by the extension should be enabled by extensions. Instructions and other enumerants should only be enabled by capability.

**Review by alan-baker — CHANGES_REQUESTED** [review]

**Comment by KonstantinSeurer** [comment]
Is there anything else that needs to be done for merging this PR? I'd like to land it before https://gitlab.freedesktop.org/mesa/mesa/-/merge_requests/24512 .

**Review by yavn — APPROVED** [review]
Thanks, looks good for me. Not sure who can help merge it now.

**Review by bashbaug — APPROVED** [review]
LGTM, this has a few approvals now so I'll just merge it.

**Pull request merged**